taichi/transforms/simplify.cpp

```diff
@@ -677,15 +678,15 @@ class BasicBlockSimplify : public IRVisitor {
       return;
     // step 1: try weakening when a struct for is used
     if (current_struct_for && !stmt->simplified) {
-      auto &loop_vars = current_struct_for->loop_vars;
-      for (int k = 0; k < (int)loop_vars.size(); k++) {
-        auto diff = irpass::analysis::value_diff(
-            stmt->input, 0, current_struct_for->loop_vars[k]);
+      int num_loop_vars = current_struct_for->snode->num_active_indices;
```

Comment:
```suggestion
      const int num_loop_vars = current_struct_for->snode->num_active_indices;
```

Sorry, personal preference...